Document behavior around removing hosts with valid health checks (#8215)
solo-io/gloo: docs/content/guides/traffic_management/request_processing/upstream_health_checks/_index.md

As part of configuring an Upstream, Gloo Edge provides the option of adding *health checks* that periodically assess the readiness of the Upstream to receive requests. See the [Envoy documentation](https://www.envoyproxy.io/docs/envoy/v1.14.1/intro/arch_overview/upstream/health_checking#arch-overview-health-checking) for more information. 

{{% notice note %}}

> If you want to be able to remove an Upstream with a working health check from Envoy's service directory, you need to set the `ignoreHealthOnHostRemoval` setting on that Upstream.

Reviewer: ```suggestion
To remove an Upstream with a working health check from Envoy's service directory, include the `ignoreHealthOnHostRemoval` setting in the Upstream's configuration.
```
